### ISSUE TITLE
Native alert disable render.

### ISSUE THREAD

**Issue by Et7f3** [revery-ui/revery#655]
I remember to have seen a similar issue fixed but can't find it.

- Operating System: Arch Linux (uptodate)
- Revery Version: 0.28.0
- OCaml version: 4.07.1
- Native, Bytecode, or JS build: Native

- Link to github repo: example app

- Steps to reproduce: build the app click native go away (in other desktop for instance) and come back it won't render

- Actual Result: the native alert prevent render.
In the result I have switch a to a windows with firefox open and I have the fraction of web page I have open render.
![image](https://user-images.githubusercontent.com/29592775/69588243-502a4c00-0fe8-11ea-880e-86cddfdcef0d.png)
- Expected Result: The app is rendered event we have native popup.

- Additional Information:

- Search terms used:

**Comment by Et7f3** [comment]
I don't know it it is related or should live in another issue. We can't track the popup 
![image](https://user-images.githubusercontent.com/29592775/69588422-cfb81b00-0fe8-11ea-8660-4f7ee475e7d8.png) (I wanted to report this issue when I have seen the other one)

**Comment by CrossR** [comment]
Could be related to https://github.com/revery-ui/revery/pull/604 and the issue in Oni2 linked there?

**Comment by zbaylin** [comment]
I think this is because when you open a dialog, GTK waits on the main thread for the result of that dialog (what button was pressed, what file was chosen, etc.). This happens with a lot of GTK apps I've found.
